Failing cram test due to non-portable use of `echo`

On current `main` (`81526067`), I get the following test failure:

```
❭ make test
./dune.exe runtest
     patdiff (internal) (exit 1)
(cd _build/.sandbox/a8ea2e44db047f26403d8c70ed472309/default && /home/craigfe/t/dune/_opam/bin/patdiff -keep-whitespace -location-style omake -unrefined test/blackbox-tests/test-cases/merlin/allow_approximate_merlin_warn.t/run.t test/blackbox-tests/test-cases/merlin/allow_approximate_merlin_warn.t/run.t.corrected)
------ test/blackbox-tests/test-cases/merlin/allow_approximate_merlin_warn.t/run.t
++++++ test/blackbox-tests/test-cases/merlin/allow_approximate_merlin_warn.t/run.t.corrected
File "test/blackbox-tests/test-cases/merlin/allow_approximate_merlin_warn.t/run.t", line 7, characters 0-1:
 |The vendored project does not trigger a third warning
 |
 |When the root project is on dune lang <= 2.7 it does not raise a warning
 |However the non-vendored sub-dir on dune lang >= 2.8 does raise a warning
 |  $ echo "(lang dune 2.7)\n (allow_approximate_merlin)" > dune-project
 |  $ dune build @check
-|  File "notvendor/dune-project", line 2, characters 0-26:
-|  2 | (allow_approximate_merlin)
-|      ^^^^^^^^^^^^^^^^^^^^^^^^^^
-|  Warning: This field was deprecated in version 2.8 of the dune language. It is
-|  useless since the Merlin configurations are not ambiguous anymore.
+|  File "dune-project", line 1, characters 0-44:
+|  1 | (lang dune 2.7)\n (allow_approximate_merlin)
+|      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+|  Error: Invalid first line, expected: (lang <lang> <version>)
+|  [1]
make: *** [Makefile:73: test] Error 1
```

which presumably happens because the cram tests pick up an implementation of `echo` that doesn't interpret backslash escapes by default. POSIX 7 makes this [implemenation defined](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/echo.html#tag_20_37_05); on my Arch machine:

```
❭ sh -c 'echo "a\nb"'
a\nb

❭ zsh -c 'echo "a\nb"'
a
b

❭ bash -c 'echo "a\nb"'
a\nb
```

The portable solution seems to be to switch to `printf`.

--- a/test/blackbox-tests/test-cases/merlin/allow_approximate_merlin_warn.t/run.t
+++ b/test/blackbox-tests/test-cases/merlin/allow_approximate_merlin_warn.t/run.t
@@ -2,7 +2,7 @@ The vendored project does not trigger a third warning
 
 When the root project is on dune lang <= 2.7 it does not raise a warning
 However the non-vendored sub-dir on dune lang >= 2.8 does raise a warning
-  $ echo "(lang dune 2.7)\n (allow_approximate_merlin)" > dune-project
+  $ printf "(lang dune 2.7)\n (allow_approximate_merlin)" > dune-project
   $ dune build @check
   File "notvendor/dune-project", line 2, characters 0-26:
   2 | (allow_approximate_merlin)

